widget/button.go

```diff
@@ -63,13 +63,18 @@ func (b *buttonRenderer) Layout(size fyne.Size) {
 // ApplyTheme is called when the Button may need to update its look
 func (b *buttonRenderer) ApplyTheme() {
 	b.label.Color = theme.TextColor()
+	if b.button.disabled {
+		b.label.Color = theme.DisabledTextColor()
+	}
 
 	b.Refresh()
 }
 
 func (b *buttonRenderer) BackgroundColor() color.Color {
-	if b.button.Style == PrimaryButton {
+	if b.button.Style == PrimaryButton && !b.button.disabled {
 		return theme.PrimaryColor()
+	} else if b.button.disabled {
+		return theme.HoverColor()
```

Comment:
I don't think we should re-use hover colour for a disabled button.
Even if the colour happens to work it should be declared separately.
We may need a disabledButton and disabledBackground color?

Reply:
I was on the fence about this too, I'll add a disabledButton color and disable the hover effect on disabled buttons. Fixed, pushed.